app/views/contexts/section_summary_context.py

```diff
@@ -164,34 +196,90 @@ def _list_summary_element(self, summary) -> Mapping:
+                    self.section, for_list=summary["for_list"], primary=True
+                ):
+                    (
+                        primary_person_edit_block_id,
+                        edit_block_id,
+                    ) = self._get_add_or_edit_blocks_primary(primary_person_block)
 
-            if primary_person_block := self._schema.get_list_collector_for_list(
-                self.section, for_list=summary["for_list"], primary=True
-            ):
-                primary_person_edit_block_id = primary_person_block[
-                    "add_or_edit_block"
-                ]["id"]
-                edit_block_id = primary_person_block["add_or_edit_block"]["id"]
+            list_summary_context = self.list_context(
+                list_collector_block["summary"],
+                for_list=list_collector_block["for_list"],
+                return_to="section-summary",
+                edit_block_id=edit_block_id,
+                remove_block_id=remove_block_id,
+                primary_person_edit_block_id=primary_person_edit_block_id,
+            )
 
-        rendered_summary = self._placeholder_renderer.render(
-            summary, self.current_location.list_item_id
-        )
+            related_answers = (
+                self._get_related_answers(current_list, list_collector_block.get("id"))
+                if current_list
+                else None
+            )
+
+            if related_answers:
+                answer_focus = f"#{self._get_answer_id(list_collector_block)}"
 
-        list_collector_block = list_collector_block or list_collector_blocks[0]
+            answer_title = (
+                self._get_answer_title(list_collector_block)
+                if related_answers
+                else None
+            )
+
+            return {
+                "title": rendered_summary["title"],
+                "type": rendered_summary["type"],
+                "add_link": add_link,
+                "add_link_text": rendered_summary["add_link_text"],
+                "empty_list_text": rendered_summary.get("empty_list_text"),
+                "list_name": rendered_summary["for_list"],
+                "related_answers": related_answers,
+                "answer_title": answer_title,
+                "answer_focus": answer_focus,
+                **list_summary_context,
+            }
+
+        if current_list.primary_person:
+
+            primary_person_block = self._schema.get_list_collector_for_list(
+                self.section, for_list=summary["for_list"], primary=True
+            )
+
+            (
+                primary_person_edit_block_id,
+                edit_block_id,
+            ) = self._get_add_or_edit_blocks_primary(
+                primary_person_block  # type: ignore
+            )
+            # primary person block always exists at this point, type hint conflicts with schema's get_list_collector_for_list return type (Optional)
```

Comment:
Where has this come from?. Not in existing code so what is it needed for in your implementation?.

Reply:
This one is now greatly simplified after some refactoring, much more similar to what it was before nesting wise. It was extra code that wasn't needed, turns out.